bin/summarizePerformanceRst.py

```diff
@@ -123,68 +127,130 @@ def plotMetricScatter(df, name1, name2, band,
-per_tract = {}
-for tract in tracts:
-    temp = astropy.table.join(single[tract], mosaic[tract], keys=join_keys, join_type='outer')
-    temp = astropy.table.join(temp, jointcal[tract], keys=join_keys, join_type='outer')
-    print(tract, len(single[tract]), len(mosaic[tract]), len(jointcal[tract]))
-    temp['tract'] = tract  # for group_by()
-    # The single Unit column (Unit_1) is going to have values for all fields,
-    # others may not (i.e. if it wasn't measured).
-    temp.remove_columns(['Unit', 'Unit_2'])
-    temp.rename_column('Unit_1', 'Unit')
-    per_tract[tract] = temp
-
-data = astropy.table.vstack(list(per_tract.values()))
-
-filters = set(data['Filter'])
-
-df = data.to_pandas()
-df.set_index(pd.MultiIndex.from_arrays([df.Metric, df.Filter]), inplace=True)
-
-descriptions = {
-    "AM1": "repeatability (marcsec) for pairs at 5 arcmin",
-    "AM2": "repeatability (marcsec) for pairs at 20 arcmin",
-    "AF1": "outlier fraction (%) for pairs at 5 min",
-    "AF2": "outlier fraction (%) for pairs at 20 min",
-    "PA1": "repeatability (mmag) of PSF source magnitudes",
-    "PF1": "outlier fraction (%) deviating by more than PA2"
-}
-
-for filt in filters:
-    plotMetricScatter(df, "AM1", "AF1", filt, xmin=0, ymin=0)
-    plotMetricScatter(df, "AM2", "AF2", filt, xmin=0, ymin=0)
-    plotMetricScatter(df, "PA1", "PF1", filt)
-    plotMetricScatter(df, "AM1", "AF1", filt, fromSingle=True, xmin=0, ymin=0)
-    plotMetricScatter(df, "AM2", "AF2", filt, fromSingle=True, xmin=0, ymin=0)
-    plotMetricScatter(df, "PA1", "PF1", filt, fromSingle=True)
-    plotMetricScatter(df, "AM1", "AF1", filt, fromSingle=True, toMosaic=True, xmin=0, ymin=0)
-    plotMetricScatter(df, "AM2", "AF2", filt, fromSingle=True, toMosaic=True, xmin=0, ymin=0)
-    plotMetricScatter(df, "PA1", "PF1", filt, fromSingle=True, toMosaic=True)
-
-print()
-print("jointcal calibrations that exceed metrics for a given filter+tract")
-print("------------------------------------------------------------------")
-for metric in ("AM1", "AF1", "AM2", "PA1"):
-    print("Metric:", metric)
-    print("-----------")
-    test = data['Metric'] == metric
-    limit = data[test]['Design']
-    exceed = data[test]['Value_jointcal'] >= limit
-    for x in data[test][exceed]:
-        print("{} > {} for: {} {}".format(x['Value_jointcal'], x['Design'], x['Filter'], x['tract']))
+def main():
+    import argparse
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("-p", "--plot", action="store_true",
+                        help="Generate metric comparison plots.")
+    parser.add_argument("-v", "--verbose", action="store_true",
+                        help="Be more verbose when reading and computing statistics.")
+    parser.add_argument("-i", "--interactive", action="store_true",
+                        help="Open an ipdb console before exiting.")
+    args = parser.parse_args()
```

Comment:
Probably not worth changing here, but I prefer putting the argument parsing down in `__main__` and unpacking the results into kwargs in `main()`, so you can call from an interactive Python prompt or notebook if you want to.  Just a pattern to consider.

Reply:
This version is my "why would anyone ever call this except from the commandline?" version. I personally prefer the:

```
if __name__ == "__main__":
    import sys
    main(sys.argv[1:])
```

style, keeping the argparse in `main()` (or even lifting it to a `parse()` method of its own).